tools/rum-models-generator/Sources/RUMModelsGeneratorCore/Input/ObjcInterop/SwiftToObjcInteropTypeTransformer.swift

```diff
@@ -78,6 +78,17 @@ internal class SwiftToObjcInteropTypeTransformer {
                     )
                     propertyWrapper.objcInteropType = try objcInteropType(for: swiftArray)
                     return propertyWrapper
+                case let swiftDictionary as SwiftDictionary where swiftDictionary.value is SwiftPrimitiveType:
+                    let propertyWrapper = ObjcInteropPropertyWrapperManagingSwiftStructProperty(
+                        owner: objcClass,
+                        swiftProperty: swiftProperty
+                    )
+                    propertyWrapper.objcInteropType = try objcInteropType(for: swiftDictionary)
+                    return propertyWrapper
+                case let swiftDictionary as SwiftDictionary where swiftDictionary.key is SwiftEnum || swiftDictionary.value is SwiftEnum:
+                    throw Exception.unimplemented(
+                        "Objc Interop for `SwiftDictionary`s with an Enum as key or value is not supported."
+                    )
```

Comment:
Maybe it's easier to change `SwiftDictionary.value` to be `SwiftPrimitiveType` to have compile-time check for this?

Reply:
That's a good idea, and similarly `SwiftDictionary.key` should be a `SwiftPrimitiveType` for now.